<launch>
    <arg name="duckiebot" default="duckiebot" />
    <arg name="world_frame" default="world" doc="tf frame id for world frame"/>
    <arg name="vehicle_frame" default="car" doc="tf frame id for vehicle frame"/>

    <arg name="grid_res" default="0.005" doc="resolution [m] of visualization grid" />

    <param name="robot_description" command="$(find xacro)/xacro.py '$(find duckietown-intnav)/config/duckiebot.xacro'"/>
    <param name="duckies_description" command="$(find xacro)/xacro.py '$(find duckietown-intnav)/config/duckies.xacro'"/>

    <node pkg="tf" type="static_transform_publisher" name="tf_vehicle_leftwheel"
        args="0 0 0 0 0 0 1 $(arg vehicle_frame) left_wheel 100" />

    <node pkg="tf" type="static_transform_publisher" name="tf_vehicle_rightwheel"
        args="0 0 0 0 0 0 1 $(arg vehicle_frame) right_wheel 100" />

    <node pkg="tf" type="static_transform_publisher" name="tf_vehicle_duckie"
        args="0 0 0.05 0 0 0 1 $(arg vehicle_frame) duckie 100" />

    <node pkg="tf" type="static_transform_publisher" name="tf_world_duckie1"
        args="-0.3 -0.5 0 0 0 0.2588223 0.9659249 $(arg world_frame) duckie1 1000" />

    <node pkg="tf" type="static_transform_publisher" name="tf_world_duckie2"
        args="-0.3 0.5 0 0 0 0.2588223 -0.9659249 $(arg world_frame) duckie2 1000" />

    <node pkg="duckietown-intnav" type="visualization_imap.py" name="imap_visualization" output="screen" >
        <param name="duckiebot" value="$(arg duckiebot)" />
        <param name="world_frame" value="$(arg world_frame)" />
        <param name="grid_res" value="$(arg grid_res)" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find duckietown-intnav)/config/rviz.rviz" />
</launch>

